ui/src/app/tasks/tasks.service.ts

```diff
@@ -1,40 +1,192 @@
 import { Observable } from 'rxjs/Observable';
 import 'rxjs/add/operator/catch';
 import 'rxjs/add/operator/map';
+import { ErrorHandler } from "../shared/model/error-handler";
+import { Page } from '../shared/model/page';
+import { TaskExecution } from './model/task-execution';
+import { TaskDefinition } from './model/task-definition';
+import { AppInfo, AppInfoOptions } from './model/app-info';
 
 @Injectable()
 export class TasksService {
 
-  private appstUrl = '/apps';
+  private taskExecutionsUrl = '/tasks/executions';
+  private appInfoUrl = '/apps/task';
+  private taskDefinitionsUrl = '/tasks/definitions';
+  public taskExecutions: Page<TaskExecution>;
+  public taskDefinitions: Page<TaskDefinition>;
 
-  constructor(private http: Http) { }
+  constructor(private http: Http, private errorHandler: ErrorHandler) {
+    this.taskExecutions = new Page<TaskExecution>();
+    this.taskDefinitions = new Page<TaskDefinition>();
+  }
+
+  getExecutions(): Observable<Page<TaskExecution>> {
+    const params = new URLSearchParams();
+    params.append('page', this.taskExecutions.pageNumber.toString());
+    params.append('size', this.taskExecutions.pageSize.toString());
+
+    if (this.taskExecutions.filter && this.taskExecutions.filter.length > 0) {
+      params.append('search', this.taskExecutions.filter);
+    }
+    return this.http.get(this.taskExecutionsUrl, {search: params})
+      .map(this.extractPagedData.bind(this))
+      .catch(this.errorHandler.handleError);
+  }
+
+  getExecution(id: string): Observable<TaskExecution> {
+    return this.http.get(this.taskExecutionsUrl + '/' + id, {})
+      .map(this.extractData.bind(this))
+      .catch(this.errorHandler.handleError);
+  }
+
+  getAppInfo(id: string): Observable<AppInfo> {
+    const params = new URLSearchParams();
+    params.append('unprefixedPropertiesOnly', 'true');
+    return this.http.get(this.appInfoUrl + '/' + id, {search: params})
+      .map(this.extractAppInfoData.bind(this))
+      .catch(this.errorHandler.handleError);
+  }
+
+  getDefinitions(): Observable<Page<TaskDefinition>> {
+    const params = new URLSearchParams();
+    params.append('page', this.taskDefinitions.pageNumber.toString());
+    params.append('size', this.taskDefinitions.pageSize.toString());
+
+    if (this.taskDefinitions.filter && this.taskDefinitions.filter.length > 0) {
+      params.append('search', this.taskDefinitions.filter);
+    }
+    return this.http.get(this.taskDefinitionsUrl, {search: params})
+      .map(this.extractDefinitionsData.bind(this))
+      .catch(this.errorHandler.handleError);
+  }
+
+  createDefinition(definition: string, name: string) {
+    console.log('Create task definition ' + definition + ' ' + name);
+    const params = new URLSearchParams();
+    params.append('definition', definition);
+    params.append('name', name);
+    const headers = new Headers({'Content-Type': 'application/json'});
```

Comment:
Up for discussion.  Instead of declaring const headers in each function can we use 
`private readonly headers = new Headers({'Content-Type': 'application/json'});` at the class level?